cmd/kubent/main.go

```diff
@@ -91,9 +100,20 @@ func main() {
 
 	log.Info().Msg(">>> Kube No Trouble `kubent` <<<")
 	log.Info().Msgf("version %s (git sha %s)", version, gitSha)
-	log.Info().Msg("Initializing collectors and retrieving data")
 
+	log.Info().Msg("Initializing collectors and retrieving data")
 	initCollectors := initCollectors(config)
+
+	if config.TargetVersion == "" {
+		config.TargetVersion, err = getServerVersion(initCollectors)
+		if err != nil {
+			log.Fatal().Err(err).Msg("Failed to detect K8s version")
+		}
+	}
+	if config.TargetVersion != "" {
```

Comment:
uhhhh not sure about this, deal with this after you set it

Reply:
Not really, these two if conditions are completely independent. I.e. it can be empty or already set by user when we hit the first `if` block.

And after it, it can still be empty because we didn't detect it. Maybe the wording is misleading? Should probably say smth. like `Target version set to XXX...`, as it doesn't have to be detected and can be coming from command line. Would updating the log message help?

Reply:
Ok, thinking about this more I just removed the second condition. This will result in the info message being printed even if the version is `""`, but that's probably ok for now. Can adjust later as we add the filtering functionality.